composer/trainer/trainer.py

```diff
@@ -1284,8 +1284,28 @@ def __init__(
             MOSAICML_ACCESS_TOKEN_ENV_VAR,
         ) is not None and not any(isinstance(x, MosaicMLLogger) for x in loggers):
             log.info('Detected run on MosaicML platform. Adding MosaicMLLogger to loggers.')
-            mosaicml_logger = MosaicMLLogger()
+            mosaicml_logger = MosaicMLLogger(
+                analytics_data={
+                    'autoresume': autoresume,
```

Comment:
would be better to pull out the analytics dictionary instead of having it defined in two places. Also suggest definining a dataclass in the mosaicmllogger file instead of just relying on a dictionary with arbitrary keys